### DISCUSSION TITLE
Referencing Microsoft.CSharp.dll

### DISCUSSION THREAD

**Discussion by SerratedSharp** [dotnet/runtime#65806] [Q&A] [answered]
I am using a package which dynamically loads Microsoft.CSharp using Assembly.Load(new AssemblyName("Microsoft.CSharp")):
https://github.com/adoconnection/RazorEngineCore/blob/e0303849684306a31a9a7bda21ecd996b44497fd/RazorEngineCore/RazorEngineCompilationOptions.cs

The problem I'm having is this fails to load in a .NET 5 WebAssembly(Uno Bootstrap) because my WASM package doesn't include Microsoft.CSharp. 

I would usually solve this be referencing the assembly in my project which ensures it gets compiled into the WASM package.
So I added the Nuget Microsoft.CSharp package, but this doesn't actually add a DLL reference.  For example, other nuget packages typically include a Compile Time Assembly:
![image](https://user-images.githubusercontent.com/97156524/155402779-0fbad41c-5674-49f1-b43c-cc0c92527fb9.png)

I tried to add a DLL reference via Add Project Reference -> Browse and choose:
packages\microsoft.csharp\4.7.0\lib\netcore50\Microsoft.CSharp.dll

I get the error:
![image](https://user-images.githubusercontent.com/97156524/155403224-e56d6096-3cfa-471a-bd67-a6ca7cbdd6b3.png)

Any insight into why it is like this, or ideas on other troubleshooting I might try would be appreciated.

**Comment by huoyaoyuan** [comment, marked as answer]
`netcore50` isn't .NET 5. It was used for Windows 8 Store, a badly chosen name.

`Microsoft.CSharp` is an inbox library in .NET Core, and always referenced. I think you need to configure ILLink to preserve it.
